### PR TITLE
search: implement advanced search

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Updates routes to add advanced search filter.
* Fixes number of parameters for component implement RecordSearchPage.
* Closes rero/rero-ils#1824.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- https://github.com/rero/rero-ils/issues/1824

## Dependencies

My PR depends on PR(s):

* rero/rero-ils#1843
* rero/ng-core#383

## How to test?

- Test advanced and simple search on some views

## Examples
- __documents:__ `identifiedBy.value:9782412052716`
- __documents:__ `identifiedBy.type:bf\:Isbn AND identifiedBy.value:9782412052716`
- __patron:__ `city:Martigny`
- __vendors:__ `name:fnac`
- __organisation:__ `gnd.preferred_name:Deputazione`

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/routes/base-route.ts [hunk 1]
```diff
@@ -93,4 +94,37 @@ export class BaseRoute {
...
+   * Advanced search filter configuration
+   * @return Object - Search filter configuration
+   */
+  protected advancedSearchFilter() {
+    return {
+      label: _('Advanced Search'),
+      filter: 'simple',
+      value: '0',
+      disabledValue: '1',
+      // activate this part when the help is online
```
Comment (zannkukai) [inline projects/admin/src/app/routes/base-route.ts:108]: add a `TODO` to find easily this statement

File: projects/admin/src/app/routes/base-route.ts [hunk 2]
```diff
@@ -93,4 +94,37 @@ export class BaseRoute {
...
+  /**
+   * Advanced search link
+   * @return string, link of help page
+   */
+  private _advancedSearchLink() {
+    switch (this._routeToolService.translateService.currentLang) {
+      case 'fr':
+        return '/help/recherche/';
+      default:
+        return '/help/recherche/';
```
Comment (zannkukai) [inline projects/admin/src/app/routes/base-route.ts:127]:
```javascript
const defaultPath =  'help/recherche';
const searchPaths = {
  'fr': 'help/recherche'
  //TODO: Add other paths
};
return (this._routeToolService.translateService.currentLang in searchPaths)
  ? searchPaths[this._routeToolService.translateService.currentLang]
  : defaultPath;
```
Looks better for me (my own opinion)

File: projects/admin/src/app/routes/corporate-bodies-route.ts [hunk 3]
```diff
@@ -48,13 +48,11 @@ export class CorporateBodiesRoute extends BaseRoute implements RouteInterface {
...
             component: ContributionBriefComponent,
             detailComponent: ContributionDetailViewComponent,
+            searchFilters: [
+              this.advancedSearchFilter()
+            ],
             aggregationsOrder: ['sources'],
-            aggregationsExpand: ['sources'],
-            // use simple query for UI search
-            preFilters: {
-              type: 'bf:Organisation',
```
Comment (zannkukai) [inline projects/admin/src/app/routes/corporate-bodies-route.ts:55]: We lost this filter ?

Comment (Garfield-fr) [reply projects/admin/src/app/routes/corporate-bodies-route.ts:55]: This is replaced by the `searchFilters`.

**Review by zannkukai — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]
Commit proposition:

```
search: implement advanced search

* Updates routes to add an advanced search filter.
* Fixes the number of parameters for the component that implements
  RecordSearchPage.
* Closes rero/rero-ils#1824.
```

**Review by pronguen — COMMENTED** [review]
I didn't tested it in detail but it worked well :+1:

Only one comment about wording
* "Advanced Search" should be renamed in "Expert search" (because in the future we will have a real advanced search as in RERO Explore, so we need two distinct names)
* Don't put a capital letter to search: "Expert search"

**Review by benerken — APPROVED** [review]

**Review by benerken — COMMENTED** [review]
- [x] (Not blocking) The "expert search" is usable if you know the keys for your search.
The librarian needs an easy and quick way to access the list of keys.

- [ ] (Not blocking) increase the size of the search box when the librarian is in the documents.

**Comment by Garfield-fr** [comment]
> * "Advanced Search" should be renamed in "Expert search" (because in the future we will have a real advanced search as in RERO Explore, so we need two distinct names)
> * Don't put a capital letter to search: "Expert search"

Fixed

**Pull request merged**